### PR TITLE
Shadow ban

### PR DESCRIPTION
**Pull request by smile0711**

### CODE REVIEW

File: api/src/controllers/message.controller.ts [hunk 1]
```diff
@@ -37,30 +37,53 @@ class MessageController {
...
-    try {
-      const { id } = session;
-      const { body } = request.body;
-      const placeId = Number.parseInt(request.params.placeId);
-      const [messageId] = await db.message
-        .insert({
-          body,
-          member_id: id,
-          place_id: placeId,
+    const bannedwords = /(nigger)|(chinc)/i;
```
Comment (smile0711) [inline api/src/controllers/message.controller.ts:47]: Need more words to be added to list (add by putting words in parenthesis and putting vertical bar between them.

File: spa/server.js [hunk 2]
```diff
@@ -115,28 +115,35 @@ io.on('connection', async function (socket) {
...
     console.log(chatData);
     if (!chatData || !chatData.msg || typeof chatData.msg !== "string") return;
     const user = USERS.get(socket);
-    if (user?.room) {
-      webhookMessage(`${user.username} in ${user.room}`, chatData.msg);
-      io.to(user.room).emit("CHAT", {
-        username: user.username,
-        msg: chatData.msg,
-      });
+    const bannedwords = /(nigger)|(chinc)/i;
```
Comment (smile0711) [inline spa/server.js:131]: Need more words to be added to list (add by putting words in parenthesis and putting vertical bar between them.)

File: api/src/controllers/message.controller.ts (hunk 1 above)
Comment (TheMikeSanto) [reply api/src/controllers/message.controller.ts:47]: Haha, even reading the words currenly in the banlist caused me to jump a bit :)

**Comment by smile0711** [comment]
I felt bad typing those out. 
 
  On Wed, Jun 15, 2022 at 1:04 PM, Mike ***@***.***> wrote:   


@TheMikeSanto commented on this pull request.

In api/src/controllers/message.controller.ts:
>      if(validator.isEmpty(request.body.body)) {
       response.status(400).json({
         error: 'Message body is required.',
       });
       return;
     }
-    
-    try {
-      const { id } = session;
-      const { body } = request.body;
-      const placeId = Number.parseInt(request.params.placeId);
-      const [messageId] = await db.message
-        .insert({
-          body,
-          member_id: id,
-          place_id: placeId,
+    const bannedwords = /(nigger)|(chinc)/i;

Haha, even reading the words currenly in the banlist caused me to jump a bit :)

**Comment by dburleson** [comment]
@smile0711 You might want to consider a pre-built list of words, which google has used in the past: https://www.npmjs.com/package/badwords-list

**Comment by dburleson** [comment]
I've tested this code and it works. Would be ideal with the mentioned library.

**Pull request merged**